### ISSUE TITLE
Allow `snip_expand()` to keep normal mode

### ISSUE THREAD

**Issue by kylo252** [L3MON4D3/LuaSnip#377]
This is mainly useful for interop with other functions outside of luasnip.

It currently goes to insert-mode by default, need to use either of these

```lua
vim.schedule(function() -- note: doesn't work without derferring
	vim.cmd [[stopinsert!]]
end)

vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes("<Esc>", true, false, true), "n", true)
```

**Comment by L3MON4D3** [comment]
Ah, yes ty!
Could you show a snippet and describe the behaviour you'd want? (There are different options eg. jump, but don't move the cursor (just changes some state in luasnip), don't jump into the snippet at all, ...)

**Comment by kylo252** [comment]
This should hopefully help

[![asciicast](https://asciinema.org/a/WdkdAm6L2uW8fi2Upm63OXkA2.svg)](https://asciinema.org/a/WdkdAm6L2uW8fi2Upm63OXkA2)

_P.s. I only chose the todo-comment workflow because it's easy to demo 😅_

**Comment by L3MON4D3** [comment]
Ah alright, so you don't need any interactivity from the snippets whatsoever
I think internally we should just make the i(0) of the expanded snippet the active node, the jumps to/from subsequently expanded snippets would be weird otherwise.

I'll see when I can get to this, might be a while.

**Comment by L3MON4D3** [comment]
I got to it :D
Does #408 work for you?

**Comment by L3MON4D3** [comment]
closed via #408
